predicates/source/source.go

```diff
@@ -0,0 +1,94 @@
+/*
+Package source implements a custom predicate to match routes
+based on the source IP of a request.
+
+It is similar in function and usage to the header predicate but
+has explicit support for IP adresses and netmasks to conveniently
+create routes based on a whole network of adresses, like a company
+network or something similar.
+
+It is important to note, that this predicate should not be used as
+the only gatekeeper for secure endpoints. Always use proper authorization
+and authentication for access control!
+
+To enable usage of this predicate behind loadbalancers or proxies, the
+X-Forwared-For header is used to determine the source of a request if it
+is available. If the X-Forwarded-For header is not present or does not contain
+a valid source address, the source IP of the incoming request is used for
+matching.
+
+The source predicate supports one or more IP addresses with or without a netmask.
+
+Examples:
+
+    // only match requests from 1.2.3.4
+    example1: Source("1.2.3.4") -> "http://example.org"
```

Comment:
it would be nice to have a semicolon separating the definitions, even if these examples not necessarily are meant as a single document.